pkg/systemd/init.js

```diff
@@ -823,4 +827,490 @@ define([
     });
 
     $(init);
-});
+    /* Timer Creation
+     * timer_unit  contains all the user's valid inputs from create-timer modal.
+     */
+    $("#create-timer").on("click", function() {
+        timer_init();
+        $("#timer-dialog").modal("show");
+    });
+
+    $("#timer-dialog").on("click", "#timer-save-button", function() {
+        var close_modal = create_timer();
+        if (close_modal)
+            $("#timer-dialog").modal("toggle");
+    });
+
+    var timer_unit = { };
+    var repeat_array = [ ];
+    var error = false;
+    var repeat_hourly_template = $("#repeat-hourly-tmpl").html();
+    mustache.parse(repeat_hourly_template);
+    var repeat_daily_template = $("#repeat-daily-tmpl").html();
+    mustache.parse(repeat_daily_template);
+    var repeat_weekly_template = $("#repeat-weekly-tmpl").html();
+    mustache.parse(repeat_weekly_template);
+    var repeat_monthly_template = $("#repeat-monthly-tmpl").html();
+    mustache.parse(repeat_monthly_template);
+    var repeat_yearly_template = $("#repeat-yearly-tmpl").html();
+    mustache.parse(repeat_yearly_template);
+    // Removes error notification when user starts typing in the error-field.
+    $("#timer-dialog").on("keypress", ".form-control", function() {
+        $(this).removeClass("has-error");
+        if ($(this).attr("id") == "hr")
+            $("#hr-error").text("");
+        else if ($(this).attr("id") == "min")
+            $("#min-error").text("");
+        else if ($(this).attr("data-content") == "hours")
+            $(this).siblings("[data-content='hr-error']").text("");
+        else if ($(this).attr("data-content") == "minutes")
+            $(this).siblings("[data-content='min-error']").text("");
+        else
+            $(this).parents("tr").next().hide();
+    });
+    /* HACK - bootstrap datepicker positions itself incorrectly on modals
```

Comment:
Is there an upstream issue for this?
Usually we try to ensure that a HACK is only temporary: either a fix is already implemented and will be available in a future version or at least the issue is known upstream and will hopefully be fixed some day.